lib/AppFactory.php

```diff
@@ -79,14 +79,19 @@ private function setErrorMiddleware(StudipPlugin $plugin, App $app): void
         $displayErrorDetails =
             (defined('\\Studip\\ENV') && \Studip\ENV === 'development') || $GLOBALS['perm']->have_perm('root');
 
-        $errorMiddleware = $app->addErrorMiddleware($displayErrorDetails, true, true);
-
-        $errorMiddleware->setDefaultErrorHandler(Errors\DefaultErrorHandler::class);
+        $container = $app->getContainer();
 
-        // Set the Not Found Handler
-        $errorMiddleware->setErrorHandler(HttpNotFoundException::class, Errors\NotFoundHandler::class);
+        $logger = $container->has(LoggerInterface::class) ? $container->get(LoggerInterface::class) : null;
 
-        // Set the Not Allowed Handler
-        $errorMiddleware->setErrorHandler(HttpMethodNotAllowedException::class, Errors\NotAllowedHandler::class);
+        $errorMiddleware = new ErrorMiddleware(
+            callableResolver: $app->getCallableResolver(),
+            responseFactory: $app->getResponseFactory(),
+            displayErrorDetails: $displayErrorDetails,
+            logErrors: true,
+            logErrorDetails: true,
+            logger: $logger,
+            plugin: $plugin
+        );
+        $app->addMiddleware($errorMiddleware);
```

Comment:
I suggest moving the instatiation of the `ErrorMiddleware` to the DI container.
```suggestion
        $app->addMiddleware(ErrorMiddleware::class);
```